xdsl/utils/type.py

```diff
@@ -53,3 +53,26 @@ def have_compatible_shape(lhs_type: Attribute, rhs_type: Attribute) -> bool:
         dim1 == DYNAMIC_INDEX or dim2 == DYNAMIC_INDEX or dim1 == dim2
         for dim1, dim2 in zip(lhs_type.get_shape(), rhs_type.get_shape())
     )
+
+
+def are_tosa_broadcastable(t_in1: Attribute, t_in2: Attribute, t_out: Attribute):
```

Comment:
if this is tosa specific then it should be in the tosa dialect. I think maybe eventually we might want such a helper in this file, but then it shouldn't reference tosa